core/src/main/java/org/ehcache/core/spi/service/ServiceUtils.java

```diff
@@ -17,38 +17,60 @@
 package org.ehcache.core.spi.service;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
-import java.util.Collections;
 
 /**
- * ServiceUtils
+ * Helper class to find a service or service configuration matching the wanted type. Note that the class
+ * is named {@code ServiceUtils} but it would actually work with anything, not only service implementations.
  */
-public class ServiceUtils {
+public final class ServiceUtils {
 
   private ServiceUtils() {
     // No instance possible
   }
 
+  /**
+   * Find instances of {@code clazz} among the {@code instances}.
+   *
+   * @param clazz searched class
+   * @param instances instances looked at
+   * @param <T> type of the searched instances
+   * @return the list of compatible instances
+   */
   public static <T> Collection<T> findAmongst(Class<T> clazz, Collection<?> instances) {
-    return findAmongst(clazz, instances.toArray());
-  }
-
-  public static <T> Collection<T> findAmongst(Class<T> clazz, Object ... instances) {
-    Collection<T> matches = new ArrayList<>();
+    Collection<T> matches = new ArrayList<>(instances.size());
     for (Object instance : instances) {
-      if (instance != null && clazz.isAssignableFrom(instance.getClass())) {
+      if (clazz.isInstance(instance)) {
```

Comment:
It's probably cleaner to rewrite this as:
```java
return instances.stream().filter(clazz::isInstance).map(clazz::cast).collect(toList());
```

Reply:
I kept it old school on purpose but if you want cute style I can :-)

Reply:
It so much more compact doing it this way, and this isn't performance sensitive code so I have no issues if it's even a bit slower for these small collections.

Reply:
Yes. I was teasing. I did it the modern way ;-)